tokio/src/signal/unix/driver.rs

```diff
@@ -42,22 +43,41 @@ pub(super) struct Inner(());
 impl Driver {
     /// Creates a new signal `Driver` instance that delegates wakeups to `park`.
     pub(crate) fn new(park: IoDriver) -> io::Result<Self> {
+        use std::mem::ManuallyDrop;
+        use std::os::unix::io::{AsRawFd, FromRawFd};
+
         // NB: We give each driver a "fresh" reciever file descriptor to avoid
         // the issues described in alexcrichton/tokio-process#42.
         //
         // In the past we would reuse the actual receiver file descriptor and
         // swallow any errors around double registration of the same descriptor.
-        // I'm not sure if the second (failed) registration simply doesn't end up
-        // receiving wake up notifications, or there could be some race condition
-        // when consuming readiness events, but having distinct descriptors for
-        // distinct PollEvented instances appears to mitigate this.
+        // I'm not sure if the second (failed) registration simply doesn't end
+        // up receiving wake up notifications, or there could be some race
+        // condition when consuming readiness events, but having distinct
+        // descriptors for distinct PollEvented instances appears to mitigate
+        // this.
         //
         // Unfortunately we cannot just use a single global PollEvented instance
         // either, since we can't compare Handles or assume they will always
         // point to the exact same reactor.
-        let receiver = globals().receiver.try_clone()?;
-        let receiver =
-            PollEvented::new_with_ready_and_handle(receiver, mio::Ready::all(), park.handle())?;
+        //
+        // Mio 0.7 removed `try_clone()` as an API due to unexpected behavior
+        // with registering dups with the same reactor. In this case, duping is
+        // safe as each dup is registered with separate reactors **and** we
+        // only at least one dup to receive the notification.
```

Comment:
All signal registrations are held in a global data structure. As long as *some* running `SignalDriver` gets woken, it will dispatch all pending signal notifications. A potential issue could be if we only wake one driver which isn't turning for whatever reason, but I believe this is unlikely

Reply:
I believe he was mostly commenting on the missing word :)

Reply:
Indeed. This threw me off a bit when reading this comment. It makes sense but took a bit of thinking to reconstruct the intent :)